src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -391,8 +392,16 @@ object ApiOperations extends DataOperations {
     * @return query result as a map
     * */
   override def queryWithPredicates(tableName: String, query: Query)(implicit ec: ExecutionContext): Future[List[Map[String, Any]]] = {
-    if (areFieldsValid(tableName, query.fields, query.predicates.map(_.field))) {
-      runQuery(TezosDatabaseOperations.selectWithPredicates(tableName, query.fields, sanitizePredicates(query.predicates)))
+    if (areFieldsValid(tableName, (query.fields ++ query.predicates.map(_.field) ++ query.orderBy.map(_.field)).toSet)) {
+      runQuery(
+        TezosDatabaseOperations.selectWithPredicates(
+          tableName,
+          query.fields,
+          sanitizePredicates(query.predicates),
+          query.orderBy,
+          Math.min(query.limit, 100000)
```

Comment:
How would I get *all* records that are available?

Reply:
In this case there is no way to take all of the records. I thought that's what you meant by `Hard limit, at least for now, should be 100000.` in the ticket description.